camel-core/src/main/java/org/apache/camel/component/dataset/DataSetEndpoint.java

```diff
@@ -59,6 +60,8 @@
     private long preloadSize;
     @UriParam(label = "consumer", defaultValue = "1000")
     private long initialDelay = 1000;
+    @UriParam(label = "consumer,producer", defaultValue = "null")
+    private Boolean disableDataSetIndex;
```

Comment:
Use a string type, and then in the component validate its either of those 3 words. And then in the `UriParam` you can specify `enums="strict,lenient,off"`

And use `off` instead of `disabled` as its seems a better word imho.